### PR TITLE
a minor trivial fixes set

### PR DESCRIPTION
**Pull request by alexrayne**
here provided a set of fixes tat i need to apply to build native and cooja motes with WERROR strict.

also included fix on Issue #1249

### CODE REVIEW

File: os/net/mac/tsch/tsch-slot-operation.c [hunk 1]
```diff
@@ -93,9 +93,9 @@
  * of the guard time (one fourth of TSCH_DEFAULT_TS_RX_WAIT) */
 #define SYNC_IE_BOUND ((int32_t)US_TO_RTIMERTICKS(tsch_timing_us[tsch_ts_rx_wait] / 4))
 
-/* By default: check that rtimer runs at >=32kHz and use a guard time of 10us */
-#if RTIMER_SECOND < (32 * 1024)
-#error "TSCH: RTIMER_SECOND < (32 * 1024)"
+/* By default: check that rtimer runs at >=8 kHz and use a guard time of 10us */
```
Comment (atiselsts) [inline os/net/mac/tsch/tsch-slot-operation.c:96]: Why did you change this?

File: os/net/routing/rpl-lite/rpl.c [hunk 2]
```diff
@@ -163,7 +163,7 @@ rpl_set_prefix_from_addr(uip_ipaddr_t *addr, unsigned len, uint8_t flags)
   }
 
   /* Try and initialize prefix */
-  memset(&curr_instance.dag.prefix_info.prefix, 0, sizeof(rpl_prefix_t));
```
Comment (atiselsts) [inline os/net/routing/rpl-lite/rpl.c:166]: I think it's the `sizeof` argument that needs to be changed, no the first argument of memset. The goal appears to be set the IPv6 address to zero, as the other members are initialized explicitly later in the code.

Comment (alexrayne) [reply os/net/routing/rpl-lite/rpl.c:166]: this is code from `prl-classic/rpl-dag.c` :

```
int
rpl_set_prefix(rpl_dag_t *dag, uip_ipaddr_t *prefix, unsigned len)
{
  rpl_prefix_t last_prefix;
  uint8_t last_len = dag->prefix_info.length;

  if(len > 128) {
    return 0;
  }
  if(dag->prefix_info.length != 0) {
    memcpy(&last_prefix, &dag->prefix_info, sizeof(rpl_prefix_t));
  }
  memset(&dag->prefix_info.prefix, 0, sizeof(dag->prefix_info.prefix));    //<-  looks like rpl-lite need this line same?
  memcpy(&dag->prefix_info.prefix, prefix, (len + 7) / 8);

```

File: os/net/mac/tsch/tsch-slot-operation.c (hunk 1 above)
Comment (alexrayne) [reply os/net/mac/tsch/tsch-slot-operation.c:96]: This comes with my first experiments with cooja polling precise resolu resolution. I wanna use RTIMER_SECOND as resolution, but later i go another way.
Anyway, imho, this change not bad, since code shouldnt stricts it`s conditions more then realy need.
This can be droped with no any suspicions.

File: os/net/routing/rpl-lite/rpl.c (hunk 2 above)
Comment (alexrayne) [reply os/net/routing/rpl-lite/rpl.c:166]: yep, refixed it.

File: os/net/mac/tsch/tsch-slot-operation.c (hunk 1 above)
Comment (atiselsts) [reply os/net/mac/tsch/tsch-slot-operation.c:96]: Yes, please drop this commit - I think we need more justification than that!

File: arch/platform/cooja/Makefile.cooja [hunk 3]
```diff
@@ -72,5 +72,7 @@ CLEAN += COOJA.log
 CFLAGSNO = $(EXTRA_CC_ARGS) -Wall -g -I/usr/local/include -DCLASSNAME=$(CLASSNAME)
 ifeq ($(WERROR),1)
 CFLAGSNO += -Werror
+CFLAGSNO += -Wno-error=unused-function -Wno-error=format-security
```
Comment (atiselsts) [inline arch/platform/cooja/Makefile.cooja:75]: Why the change? Especially `-Wno-format-security` and `-Wno-error=format-security`. I think it should be left as is.
Which code caused you problems to build without these flags?

Comment (alexrayne) [reply arch/platform/cooja/Makefile.cooja:75]: as i remember printf with only one fmt, whitout arguments, causes this warning

**Comment by alexrayne** [comment]
close for cleanup PR
